growth-ring/src/wal.rs

```diff
@@ -59,8 +61,23 @@ type WalFileId = u64;
 pub type WalBytes = Box<[u8]>;
 pub type WalPos = u64;
 
-fn get_fid(fname: &str) -> WalFileId {
-    scan_fmt!(fname, "{x}.log", [hex WalFileId]).unwrap()
+// convert XXXXXX.log into number from the XXXXXX (in hex)
+fn get_fid(fname: &Path) -> Result<WalFileId, WalError> {
+    let wal_err: WalError = WalError::Other("not a log file".to_string());
```

Comment:
In theory, this is an allocation with every call to this function. I think it would be better to repeat the error creation.